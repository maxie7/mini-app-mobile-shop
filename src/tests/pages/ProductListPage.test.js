import React from "react";
import {shallow} from "enzyme";
import ProductListPage from "../../pages/ProductList/ProductListPage.component";

describe('ProductListPage component', () => {
  it('should render component (Product List Page)', () => {
    expect(shallow(<ProductListPage />)).toMatchSnapshot();
  })

  it('should have Searcher component', () => {
    const component = shallow(<ProductListPage />);
    expect(component.find('.Searcher')).toBeTruthy();
  })

  it('should have ProductListPageContainer wrapper', () => {
    const component = shallow(<ProductListPage />);
    expect(component.find('.ProductListPageContainer')).toBeTruthy();
  })

  it('should have at least one MobileItem component', () => {
    const component = shallow(<ProductListPage />);
    expect(component.find('.MobileItem')).toBeTruthy();
  })
})

// TODO: we need to write a test for filteredMobiles method and its behavior
// describe('Filtering content with search input', () =>{
//
//   let wrapper;
//   const setState = jest.fn();
//   const useStateSpy = jest.spyOn(React, "useState")
//   useStateSpy.mockImplementation((init) => [init, setState]);
//
//   beforeEach(() => {
//     wrapper = mount(shallow(<ProductListPage />).get(0))
//   });
//
//   afterEach(() => {
//     jest.clearAllMocks();
//   });
//
//   it('should show product which match the search input field', () => {
//
//   })
// })