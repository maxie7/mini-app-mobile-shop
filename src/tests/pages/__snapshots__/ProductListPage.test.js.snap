// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListPage component should render component (Product List Page) 1`] = `ShallowWrapper {}`;
